src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointResponse.cs

```diff
@@ -84,18 +83,19 @@ private bool GetIsIResult()
             SymbolEqualityComparer.Default.Equals(ResponseType, resultType);
     }
 
-    private string? GetContentType(IMethodSymbol method)
+    private string? GetContentType()
     {
         // `void` returning methods do not have a Content-Type.
         // We don't have a strategy for resolving a Content-Type
         // from an IResult. Typically, this would be done via an
         // IEndpointMetadataProvider so we don't need to set a
         // Content-Type here.
-        if (method.ReturnsVoid || IsIResult)
+        if (IsIResult || HasNoResponse)
         {
             return null;
         }
-        return method.ReturnType.SpecialType is SpecialType.System_String ? "text/plain" : "application/json";
+
+        return ResponseType!.SpecialType is SpecialType.System_String ? "text/plain; charset=utf-8" : "application/json";
```

Comment:
charset on json?

Reply:
Yeah just followed through the RDF code path and you are correct. Presumably a few of the future tests we migrate will pick this up.

Reply:
Turns out this is not the case. I just migrated the test case across into the shared infrastructure so I could compare RDF vs. RDG output and the response is actually meant to be `application/json`.

I'll leave it as it is and hopefully if there some more nuance to this a future testcase we migrate will highlight it.